frame/aura/src/lib.rs

```diff
@@ -179,8 +187,12 @@ impl<T: Config> OneSessionHandler<T::AccountId> for Pallet<T> {
 		if changed {
 			let next_authorities = validators.map(|(_, k)| k).collect::<Vec<_>>();
 			let last_authorities = Self::authorities();
-			if next_authorities != last_authorities {
-				Self::change_authorities(next_authorities);
+			if last_authorities != next_authorities {
+				let bounded = match next_authorities.try_into() {
+					Ok(v) => v,
+					Err(_) => return,
```

Comment:
we should at least log an error, on_new_session shouldn't really be allowed to fail to set the authorities.
I don't know what is consequences of this if it happens.

Maybe better to use a WeakBoundedVec no ?

Reply:
In #8640, what we did instead is to truncate the authorities so that we're only left with the first n number of authorities, where n is the value of `T::MaxAuthorities`. I guess a WeakBoundedVec would work here as well -- it'll just mean that no additional authorities can be inserted, a removal must happen in order to reduce the number back down to n or below it.